mail_store_outgoing/models/ir_mail_server.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+import logging
+import re
+import imaplib
+from odoo import tools
+from odoo.exceptions import ValidationError
+from odoo import fields
+from odoo import models
+from odoo import api
+from odoo.tools.translate import _
+
+_logger = logging.getLogger(__name__)
+
+
+class IrMailServer(models.Model):
+    _inherit = 'ir.mail_server'
+
+    def parse_list_response(self, line):
+        list_response_pattern = re.compile(
+            r'\((?P<flags>.*?)\) "(?P<delimiter>.*)" (?P<name>.*)'
+        )
+        flags, delimiter, mailbox_name = \
+            list_response_pattern.match(line).groups()
+        mailbox_name = mailbox_name.strip('"')
+        return (flags, delimiter, mailbox_name)
+
+    @api.multi
+    def test_imap_connection(self):
+        self.ensure_one()
+        imap_pool = self.env['ir.mail.imap.folder']
+        if self.has_separate_imap_server:
+            smtp_server = self.separate_imap_server
+        else:
+            smtp_server = self.smtp_host
+        try:
+            maillib = imaplib.IMAP4_SSL(smtp_server)
+            result = maillib.login(self.smtp_user, self.smtp_pass)
+            typ, mBoxes = maillib.list()
+            folder_ids = imap_pool.search([('server_id', '=', self.id)])
+            for folder in folder_ids:
+                folder.unlink()
+            for line in mBoxes:
+                flags, delimiter, mailbox_name = self.parse_list_response(line)
+                res = {
+                    'server_id': self.id,
+                    'name': mailbox_name,
+                }
+                imap_pool.create(res)
+            self.write({'imap_mailbox_verified': True})
+        except Exception, e:
+            raise ValidationError(
+                _("Connection Test Failed!"),
```

Comment:
ValidationError only needs one parameter. 
See https://github.com/odoo/odoo/blob/10.0/odoo/exceptions.py#L73

Reply:
Okay i will fix it today